Library/String+CurrencyTests.swift

```diff
@@ -0,0 +1,409 @@
+import XCTest
+@testable import Library
+
+// swiftlint:disable line_length
+final class StringCurrencyTests: XCTestCase {
+  func testCurrencySymbol_NoSymbol() {
+    let currencySymbol = ""
+    let attributedString = attributedCurrencyString(
+      currencySymbol: currencySymbol,
+      amount: 99.975,
+      fractionDigits: 2,
+      font: UIFont.ksr_title1(),
+      superscriptFont: UIFont.ksr_body(),
+      foregroundColor: UIColor.cyan
+    )
+    let range = NSRange(location: 0, length: currencySymbol.count)
+    let attributedSubstring = attributedString.attributedSubstring(from: range)
+
+    XCTAssertEqual("99.97", attributedString.string)
+    XCTAssertEqual(0, attributedSubstring.length)
+  }
+
+  func testCurrencySymbol_SimpleSymbol() {
+    let currencySymbol = "$"
+    let font = UIFont.ksr_title1()
+    let superscriptFont = UIFont.ksr_body()
+    let foregroundColor = UIColor.cyan
+    let attributedString = attributedCurrencyString(
+      currencySymbol: currencySymbol,
+      amount: 99.975,
+      fractionDigits: 2,
+      font: font,
+      superscriptFont: superscriptFont,
+      foregroundColor: foregroundColor
+    )
+    let range = NSRange(location: 0, length: currencySymbol.count)
+    let attributedSubstring = attributedString.attributedSubstring(from: range)
+    let attributes = attributedSubstring.attributes(at: 0, longestEffectiveRange: nil, in: range)
+    let offset = font.capHeight - superscriptFont.capHeight
+
+    XCTAssertEqual("$99.97", attributedString.string)
+    XCTAssertEqual(currencySymbol, attributedSubstring.string)
+    XCTAssertEqual(superscriptFont, attributes[NSAttributedString.Key.font] as? UIFont)
+    XCTAssertEqual(foregroundColor, attributes[NSAttributedString.Key.foregroundColor] as? UIColor)
+    XCTAssertEqual(offset, (attributes[NSAttributedString.Key.baselineOffset] as? NSNumber) as? CGFloat)
```

Comment:
Interesting that it needs the cast to `NSNumber` first 🤔

Reply:
What was I even thinking? 😄